public/main.ts

```diff
@@ -132,27 +132,196 @@ const kingdoms = [
 const player = kingdoms[0];
 player.farms.set(Crop.Rye, new Farm(getRandInt(minStartingRye, maxStartingRye)));
 
+// indicies for select fields
+let selectedFarm = "Rye";
+// let selectedCrop = "Rye";
+// let sellAmount = "";
+
 function promptPlayer(message: string): string {
     let result = null;
     while(result == null)
         result = window.prompt(message, "");
     return result;
 }
 
-function updateText() {
-
+interface Printable { toString(): string }
+type tag = HTMLElement | HTMLInputElement;
+function setElementInnerHTML(el: tag, content: Printable) {
+    el.innerHTML = content.toString();
 }
+function setIdInnerHTML(id: string, content: Printable) {
+    document.getElementById(id)!.innerHTML = content.toString();
```

Comment:
This could be written in terms of the `setElementInnerHTML` function.
```suggestion
    setElementInnerHTML(document.getElementById(id)!, content);
```